sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/components/EmailGenerator.groovy

```diff
@@ -3,65 +3,99 @@ package life.qbic.samplenotificator.components
-            prepareEmailTemplate(templatePath)
-            prepareEmails(notificationContent)
+            Document filledEmailContent = fillEmailContent(notificationContent)
+            File emailHTMLFile = GroupEmailBodyAndHeaderIntoHTMLFile(filledEmailContent.html())
+            send(emailHTMLFile, notificationContent.customerEmailAddress)
         }
     }
 
     /**
-     * Reads in the HTML Template provided TemplatePath
-     * containing the provided notifications to the provided subscribers
+     * Loads the information stored in the Header and EmailTemplate File into Suppliers
+     * since the underlying InputStreams are consumed upon first time usage.
      */
-    private void prepareEmailTemplate(String templatePath) {
+    private void loadEmailTemplates() {
         //Template Content is stored in Jar and can only be accessed via InputStream which is consumed for each Jsoup Parsing
-        InputStream EMAIL_HTML_TEMPLATE_STREAM = EmailHTMLTemplate.class.getClassLoader().getResourceAsStream(templatePath)
-        this.emailHTMLTemplate = new EmailHTMLTemplate(Jsoup.parse(EMAIL_HTML_TEMPLATE_STREAM, "UTF-8",""))
+        emailHeaderTemplateSupplier = () -> EmailHTMLTemplate.class.getClassLoader().getResourceAsStream(emailHeaderPath)
+        emailNotificationTemplateSupplier = ()  -> EmailHTMLTemplate.class.getClassLoader().getResourceAsStream(emailTemplatePath)
     }
 
     /**
-     * Triggers the creation and submission of each email
-     * containing the provided notifications to the provided subscribers
+     * Triggers the adaption of the emailTemplate to contain the information provided in the notificationContent
      */
-    private void prepareEmails(NotificationContent notificationContent) {
+    private Document fillEmailContent(NotificationContent notificationContent) {
+        emailHTMLTemplate = new EmailHTMLTemplate(Jsoup.parse(emailNotificationTemplateSupplier.get(), "UTF-8", ""))
         Document filledEmailTemplate = emailHTMLTemplate.fillTemplate(notificationContent)
-        emails.put(filledEmailTemplate, notificationContent.customerEmailAddress)
+        return filledEmailTemplate
+    }
+
+    /**
+     * Concatenates email header information and the actual email content into a singular file
+     *
+     * Method to create a file containing the email header information{@see notification-template/header.txt} and the dynamically adapted email message.
+     * This is necessary since sendmail doesn't provide a dedicated command line switch to set the content-type of the message
+     * @param notificationContent String representation of the filled in HTML Email Template {@see notification-template/email-update-template.html}
+     * @return concatenatedHTMLFile of the mailutils sendmail tool for detailed information see(@link <a href=https://www.cs.ait.ac.th/~on/O/oreilly/tcpip/sendmail/ch36_05.htm/>here</a>)
+     */
+    private File GroupEmailBodyAndHeaderIntoHTMLFile(String notificationContent) {
```

Comment:
Is there a reason the method name starts in upercase?

Reply:
Upper Case Is Best Case 💪